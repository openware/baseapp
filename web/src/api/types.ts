declare global {
    interface Config {
        api: {
            authUrl: string;
            tradeUrl: string;
            applogicUrl: string;
            rangerUrl: string;
            finexUrl: string;
            p2pUrl: string;
        };
        finex: string | boolean;
        withCredentials: string | boolean;
        incrementalOrderBook: string | boolean;
        isResizable: string | boolean;
        isDraggable: string | boolean;
        showLanding: string | boolean;
        sentryEnabled: string | boolean;
        captchaLogin: string | boolean;
        usernameEnabled: string | boolean;
        gaTrackerKey: string;
        minutesUntilAutoLogout: string;
        msAlertDisplayTime: string;
        msPricesUpdates: string;
        sessionCheckInterval: string;
        balancesFetchInterval: string;
        passwordEntropyStep: string | number;
        storage: {
            defaultStorageLimit: string | number;
            orderBookSideLimit: string | number;
        };
        languages: string[];
        kycSteps: string[];
        captcha_type: 'recaptcha' | 'geetest' | 'none';
        captcha_id?: string;
        password_min_entropy: string | number;
        palette?: string;
        wizard_step?: string;
        barong_upload_size_min_range?: string;
        barong_upload_size_max_range?: string;
        themeSwitcher: 'visible' | 'hidden';
        peatio_platform_currency?: string;
        useSharedLayout: string;
        footerNavigation?: string;
        footerSocials?: string;
        navigations?: string;
        logo_icons?: string;
        fontFamily?: string;
    }

    interface Window {
        env: Config;
    }
}

export {};
