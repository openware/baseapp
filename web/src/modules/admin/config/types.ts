export interface ConfigUpdateDataInterface {
    scope: string;
    component: string;
    key: string;
    value: string;
}
