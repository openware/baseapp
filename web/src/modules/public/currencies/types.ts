export interface Currency {
    id: string;
    name: string;
    symbol: string;
    explorer_transaction: string;
    explorer_address: string;
    type: string;
    deposit_fee: string;
    min_confirmations: number;
    min_deposit_amount: string;
    withdraw_fee: string;
    min_withdraw_amount: string;
    withdraw_limit_24h: string;
    withdraw_limit_72h: string;
    deposit_enabled: boolean;
    withdrawal_enabled: boolean;
    base_factor: number;
    precision: number;
    icon_url: string;
}
