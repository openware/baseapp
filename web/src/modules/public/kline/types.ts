export interface KlineEvent {
    time: number;
    close: number;
    open: number;
    high: number;
    low: number;
    volume: number;
}
