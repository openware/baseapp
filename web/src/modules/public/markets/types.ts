export type MarketId = string;

export interface MarketFilterCustomStepRule {
    limit: string;
    step: string;
}

export interface MarketFilterCustomStep {
    type: string;
    rules: MarketFilterCustomStepRule[];
}

export interface MarketFilterSignificantDigit {
    type: string;
    digits: number;
}

export type MarketFilter = MarketFilterSignificantDigit | MarketFilterCustomStep;

export interface Market {
    id: MarketId;
    name: string;
    base_unit: string;
    quote_unit: string;
    min_price: string;
    max_price: string;
    min_amount: string;
    amount_precision: number;
    price_precision: number;
    state?: string;
    filters?: MarketFilter[];
}

export interface Ticker {
    amount: string;
    avg_price: string;
    high: string;
    last: string;
    low: string;
    open: number | string;
    price_change_percent: string;
    volume: string;
}

export interface TickerEvent {
    amount: string;
    name: string;
    base_unit: string;
    quote_unit: string;
    low: string;
    high: string;
    open: number;
    last: string;
    avg_price: string;
    price_change_percent: string;
    volume: string;
    at: number;
}

export interface MarketPriceInterface {
    price: string;
    created_at: string;
    updated_at: string;
}
