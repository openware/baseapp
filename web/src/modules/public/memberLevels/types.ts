export interface MemberLevels {
    deposit: { minimum_level: number };
    withdraw: { minimum_level: number };
    trading: { minimum_level: number };
}
