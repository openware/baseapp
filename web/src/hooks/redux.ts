import { shallowEqual, useSelector } from 'react-redux';
import { RootState } from '../modules';

export function useReduxSelector<T>(selector: (state: RootState) => T): T {
    return useSelector(selector, shallowEqual);
}
