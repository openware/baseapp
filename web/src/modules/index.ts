import { combineReducers } from 'redux';
import { all, call } from 'redux-saga/effects';
import { adminReducer, publicReducer, userReducer } from './app';
import { ConfigUpdateState, rootConfigUpdateSaga } from './admin/config';
import { AlertState, rootHandleAlertSaga } from './public/alert';
import { BlockchainsState } from './public/blockchains';
import { BlocklistAccessState, rootBlocklistAccessSaga } from './public/blocklistAccess';
import { ConfigsState, rootConfigsSaga } from './public/configs';
import { CurrenciesState } from './public/currencies';
import { ErrorHandlerState, rootErrorHandlerSaga } from './public/errorHandler';
import { GlobalSettingsState } from './public/globalSettings';
import { GridLayoutState } from './public/gridLayout';
import { LanguageState } from './public/i18n';
import { KlineState, rootKlineFetchSaga } from './public/kline';
import { MarketsState, rootMarketsSaga } from './public/markets';
import { MemberLevelsState, rootMemberLevelsSaga } from './public/memberLevels';
import { DepthIncrementState, DepthState, OrderBookState, rootOrderBookSaga } from './public/orderBook';
import { RecentTradesState, rootRecentTradesSaga } from './public/recentTrades';
import { ApiKeysState } from './user/apiKeys';
import { rootApiKeysSaga } from './user/apiKeys/sagas';
import { AuthState, rootAuthSaga } from './user/auth';
import { AbilitiesState, rootAbilitiesSaga } from './user/abilities';
import { BeneficiariesState, rootBeneficiariesSaga } from './user/beneficiaries';
import { GeetestCaptchaState, rootGeetestCaptchaSaga } from './user/captcha';
import { DocumentationState, rootDocumentationSaga } from './user/documentation';
import { EmailVerificationState, rootEmailVerificationSaga } from './user/emailVerification';
import { HistoryState, rootHistorySaga } from './user/history';
import { InternalTransfersState, rootInternalTransfersSaga } from './user/internalTransfers';
import { AddressesState, rootSendAddressesSaga } from './user/kyc/addresses';
import { DocumentsState, rootSendDocumentsSaga } from './user/kyc/documents';
import { IdentityState, rootSendIdentitySaga } from './user/kyc/identity';
import { LabelState, rootLabelSaga } from './user/kyc/label';
import { PhoneState, rootSendCodeSaga } from './user/kyc/phone';
import { OpenOrdersState, rootOpenOrdersSaga } from './user/openOrders';
import { OrdersState, rootOrdersSaga } from './user/orders';
import { OrdersHistoryState, rootOrdersHistorySaga } from './user/ordersHistory';
import { PasswordState, rootPasswordSaga } from './user/password';
import { ProfileState, rootProfileSaga } from './user/profile';
import { rootUserActivitySaga, UserActivityState } from './user/userActivity';
import { rootWalletsSaga, WalletsState } from './user/wallets';
import { QuickExchangeState, rootQuickExchangeSaga } from './user/quickExchange';
import { rootWithdrawLimitSaga, WithdrawLimitState } from './user/withdrawLimit';
import { MarketsAdminState, rootMarketsAdminSaga } from './admin/markets';
import { PlatformCreateState, rootPlatformCreateSaga } from './admin/platform';
import { P2PState, rootP2PSaga } from './public/p2p';
import { PaymentMethodState, rootPaymentMethodSaga } from './user/paymentMethod';
import { P2POffersState, rootP2POffersSaga } from './user/p2pOffers';
import { P2PTransfersState, rootP2PTransfersSaga } from './user/p2pTransfers';
import { P2POrdersState, rootP2POrdersSaga } from './user/p2pOrders';
import { P2PDisputeState, rootP2PDisputeSaga } from './user/p2pDispute';
import { rootFeeGroupSaga, FeeGroupState } from './user/feeGroup';
import { rootWithdrawLimitsSaga, WithdrawLimitsState } from './public/withdrawLimits';

export * from './admin/config';
export * from './admin/markets';
export * from './admin/platform';
export * from './public/alert';
export * from './public/blockchains';
export * from './public/blocklistAccess';
export * from './public/configs';
export * from './public/currencies';
export * from './public/errorHandler';
export * from './public/globalSettings';
export * from './public/i18n';
export * from './public/kline';
export * from './public/markets';
export * from './public/memberLevels';
export * from './public/orderBook';
export * from './public/recentTrades';
export * from './public/p2p';
export * from './public/withdrawLimits';
export * from './user/apiKeys';
export * from './user/auth';
export * from './user/beneficiaries';
export * from './user/captcha';
export * from './user/documentation';
export * from './user/emailVerification';
export * from './user/history';
export * from './user/internalTransfers';
export * from './user/kyc';
export * from './user/openOrders';
export * from './user/orders';
export * from './user/ordersHistory';
export * from './user/password';
export * from './user/profile';
export * from './user/userActivity';
export * from './user/wallets';
export * from './user/feeGroup';
export * from './user/withdrawLimit';
export * from './user/quickExchange';
export * from './user/abilities';
export * from './user/paymentMethod';
export * from './user/p2pOffers';
export * from './user/p2pOrders';
export * from './user/p2pTransfers';
export * from './user/p2pDispute';

export interface RootState {
    public: {
        alerts: AlertState;
        blockchains: BlockchainsState;
        blocklistAccess: BlocklistAccessState;
        globalSettings: GlobalSettingsState;
        configs: ConfigsState;
        currencies: CurrenciesState;
        depth: DepthState;
        errorHandler: ErrorHandlerState;
        i18n: LanguageState;
        incrementDepth: DepthIncrementState;
        kline: KlineState;
        markets: MarketsState;
        memberLevels: MemberLevelsState;
        orderBook: OrderBookState;
        recentTrades: RecentTradesState;
        rgl: GridLayoutState;
        p2p: P2PState;
        withdrawLimits: WithdrawLimitsState,
    };
    user: {
        abilities: AbilitiesState;
        addresses: AddressesState;
        apiKeys: ApiKeysState;
        auth: AuthState;
        beneficiaries: BeneficiariesState;
        captcha: GeetestCaptchaState;
        documentation: DocumentationState;
        documents: DocumentsState;
        history: HistoryState;
        identity: IdentityState;
        internalTransfers: InternalTransfersState;
        label: LabelState;
        openOrders: OpenOrdersState;
        orders: OrdersState;
        ordersHistory: OrdersHistoryState;
        password: PasswordState;
        phone: PhoneState;
        profile: ProfileState;
        sendEmailVerification: EmailVerificationState;
        userActivity: UserActivityState;
        wallets: WalletsState;
        withdrawLimit: WithdrawLimitState;
        feeGroup: FeeGroupState;
        quickExchange: QuickExchangeState;
        paymentMethod: PaymentMethodState;
        p2pOffers: P2POffersState;
        p2pTransfers: P2PTransfersState;
        p2pOrders: P2POrdersState;
        p2pDispute: P2PDisputeState;
    };
    admin: {
        configUpdate: ConfigUpdateState;
        markets: MarketsAdminState;
        platform: PlatformCreateState;
    };
}

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
    admin: adminReducer,
});

export function* rootSaga() {
    yield all([
        call(rootWithdrawLimitsSaga),
        call(rootFeeGroupSaga),
        call(rootQuickExchangeSaga),
        call(rootAbilitiesSaga),
        call(rootApiKeysSaga),
        call(rootAuthSaga),
        call(rootBeneficiariesSaga),
        call(rootBlocklistAccessSaga),
        call(rootConfigUpdateSaga),
        call(rootDocumentationSaga),
        call(rootEmailVerificationSaga),
        call(rootErrorHandlerSaga),
        call(rootGeetestCaptchaSaga),
        call(rootHandleAlertSaga),
        call(rootHistorySaga),
        call(rootInternalTransfersSaga),
        call(rootKlineFetchSaga),
        call(rootLabelSaga),
        call(rootMarketsAdminSaga),
        call(rootMarketsSaga),
        call(rootMemberLevelsSaga),
        call(rootOpenOrdersSaga),
        call(rootOrderBookSaga),
        call(rootOrdersHistorySaga),
        call(rootP2PSaga),
        call(rootOrdersSaga),
        call(rootPasswordSaga),
        call(rootPlatformCreateSaga),
        call(rootProfileSaga),
        call(rootRecentTradesSaga),
        call(rootSendAddressesSaga),
        call(rootSendCodeSaga),
        call(rootSendDocumentsSaga),
        call(rootSendIdentitySaga),
        call(rootUserActivitySaga),
        call(rootWalletsSaga),
        call(rootWithdrawLimitSaga),
        call(rootQuickExchangeSaga),
        call(rootPaymentMethodSaga),
        call(rootP2POffersSaga),
        call(rootP2POrdersSaga),
        call(rootP2PTransfersSaga),
        call(rootP2PDisputeSaga),
        call(rootConfigsSaga),
    ]);
}
