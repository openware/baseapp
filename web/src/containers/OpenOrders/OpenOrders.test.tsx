import { shallow } from 'enzyme';
import * as React from 'react';
import { connect, Provider } from 'react-redux';
import { createStore } from 'redux';
import { OpenOrdersComponent, OpenOrdersProps } from '..';
import { Market, rootReducer } from '../../modules';
import { OrderCommon } from '../../modules/types';

const currentMarket: Market | undefined = {
    id: 'ethusd',
    name: 'ETH/USD',
    base_unit: 'eth',
    quote_unit: 'usd',
    min_price: '0.0',
    max_price: '0.0',
    min_amount: '0.0',
    amount_precision: 4,
    price_precision: 4,
};

const openOrdersData: OrderCommon[] = [
    {
        id: 131,
        side: 'sell',
        price: '104.4313',
        created_at: '2019-01-31T21:14:04+01:00',
        remaining_volume: '0',
        origin_volume: '10',
        executed_volume: '10',
        state: 'wait',
        market: 'ethusd',
    },
];

const defaultProps: OpenOrdersProps = {
    currentMarket,
    list: openOrdersData,
    fetching: false,
    cancelFetching: false,
    userLoggedIn: true,
};

const store = createStore(rootReducer);
const OpenOrders = connect()(OpenOrdersComponent);

const setup = (props: Partial<OpenOrdersProps> = {}) =>
    shallow(
        <Provider store={store}>
            // @ts-ignore
            <OpenOrders {...{ ...defaultProps, ...props }} />
        </Provider>,
    );

describe('OpenOrders', () => {
    let wrapper = setup();

    beforeEach(() => {
        wrapper = setup();
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
