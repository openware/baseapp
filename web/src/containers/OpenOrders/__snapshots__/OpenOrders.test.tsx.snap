// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenOrders should render 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  // @ts-ignore
  <Connect(Component)
    cancelFetching={false}
    currentMarket={
      Object {
        "amount_precision": 4,
        "base_unit": "eth",
        "id": "ethusd",
        "max_price": "0.0",
        "min_amount": "0.0",
        "min_price": "0.0",
        "name": "ETH/USD",
        "price_precision": 4,
        "quote_unit": "usd",
      }
    }
    fetching={false}
    list={
      Array [
        Object {
          "created_at": "2019-01-31T21:14:04+01:00",
          "executed_volume": "10",
          "id": 131,
          "market": "ethusd",
          "origin_volume": "10",
          "price": "104.4313",
          "remaining_volume": "0",
          "side": "sell",
          "state": "wait",
        },
      ]
    }
    userLoggedIn={true}
  />
</ContextProvider>
`;
